op-chain-ops/crossdomain/migrate.go

```diff
@@ -0,0 +1,112 @@
+package crossdomain
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-bindings/predeploys"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+)
+
+var (
+	abiTrue                      = common.Hash{31: 0x01}
+	errLegacyStorageSlotNotFound = errors.New("cannot find storage slot")
+)
+
+// This takes a state db and a list of withdrawals
+func MigrateWithdrawals(withdrawals []*PendingWithdrawal, db vm.StateDB) error {
+	for _, legacy := range withdrawals {
+		legacySlot, err := legacy.StorageSlot()
+		if err != nil {
+			return err
+		}
+
+		legacyValue := db.GetState(predeploys.LegacyMessagePasserAddr, legacySlot)
+		if legacyValue != abiTrue {
+			return fmt.Errorf("%w: %s", errLegacyStorageSlotNotFound, legacyValue)
+		}
+
+		// TODO: pass in args
+		withdrawal, err := MigrateWithdrawal(&legacy.LegacyWithdrawal, &common.Address{}, &common.Address{})
+		if err != nil {
+			return err
+		}
+
+		slot, err := withdrawal.StorageSlot()
+		if err != nil {
+			return err
+		}
+
+		db.SetState(predeploys.L2ToL1MessagePasserAddr, slot, abiTrue)
+	}
+	return nil
+}
+
+// MigrateWithdrawal will turn a LegacyWithdrawal into a bedrock
+// style Withdrawal.
+func MigrateWithdrawal(withdrawal *LegacyWithdrawal, l1CrossDomainMessenger, l1StandardBridge *common.Address) (*Withdrawal, error) {
+	value := new(big.Int)
+
+	isFromL2StandardBridge := *withdrawal.Sender == predeploys.L2StandardBridgeAddr
+	isToL1StandardBridge := *withdrawal.Target == *l1StandardBridge
```

Comment:
Is target guaranteed to be non-nil?

Reply:
Assuming there is no adversarial input, yes target will always be non-nil. I can double check it